cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -2535,6 +2535,12 @@ if(ARROW_S3)
   include_directories(SYSTEM ${AWSSDK_INCLUDE_DIR})
   message(STATUS "Found AWS SDK headers: ${AWSSDK_INCLUDE_DIR}")
   message(STATUS "Found AWS SDK libraries: ${AWSSDK_LINK_LIBRARIES}")
+
+  if(APPLE)
+    set_target_properties(AWS::aws-c-common
```

Comment:
Can you add a comment here?

Also, can you report the issue upstream?

Reply:
Sure

Reply:
Reported upstream https://github.com/aws/aws-sdk-cpp/issues/1245